From efbf22730b90cf127812e1026f3a2947d5fe643e Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <natechancellor@gmail.com>
Date: Wed, 3 Oct 2018 14:31:35 -0700
Subject: [PATCH 07/13] DO-NOT-UPSTREAM: Disable -Wswitch

These warnings are related to use of the _IOC macro to define certain
values for switch statements which then overflow because the type is int
when it really should be unsigned int. There will need to be some
discussion about how to solve that in the scsi subsystem and until that
happens, this is just noise.

Link: https://github.com/ClangBuiltLinux/linux/issues?q=label:-Wswitch
Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 19a6fac3d880..c99bb787f7ae 100644
--- a/Makefile
+++ b/Makefile
@@ -711,6 +711,7 @@ KBUILD_CFLAGS += $(call cc-disable-warning, address-of-packed-member)
 KBUILD_CFLAGS += $(call cc-disable-warning, tautological-compare)
 KBUILD_CFLAGS += $(call cc-disable-warning, duplicate-decl-specifier)
 KBUILD_CFLAGS += $(call cc-disable-warning, empty-body)
+KBUILD_CFLAGS += $(call cc-disable-warning, switch)
 # CLANG uses a _MergedGlobals as optimization, but this breaks modpost, as the
 # source of a reference will be _MergedGlobals and not on of the whitelisted names.
 # See modpost pattern 2
-- 
2.19.0

